src/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -181,6 +197,218 @@ private SymbolTable getSystemSymbolTable() {
         return SharedSymbolTable.getSystemSymbolTable(getIonMajorVersion());
     }
 
+    /**
+     * Read-only snapshot of the local symbol table at the reader's current position.
+     */
+    private class LocalSymbolTableSnapshot implements SymbolTable, SymbolTableAsStruct {
+
+        // The system symbol table.
+        private final SymbolTable system = IonReaderContinuableApplicationBinary.this.getSystemSymbolTable();
+
+        // The max ID of this local symbol table.
+        private final int maxId;
+
+        // The shared symbol tables imported by this local symbol table.
+        private final LocalSymbolTableImports importedTables;
+
+        // Map representation of this symbol table. Keys are symbol text; values are the lowest symbol ID that maps
+        // to that text.
+        final Map<String, Integer> mapView;
+
+        // List representation of this symbol table, indexed by symbol ID.
+        final String[] listView;
```

Comment:
naming these for what they do (e.g. `textToId` and `idToText`) is clearer than what data structure is used to hold the mapping.

Reply:
Done, to be included in a separate PR.

Reply:
Addressed in https://github.com/amazon-ion/ion-java/pull/539